### ISSUE TITLE
Error in call to extension function  com.occamlab.te.TECore - hard to debug

### ISSUE THREAD

**Issue by bermud** [opengeospatial/teamengine#101]
Make changes in the code so better error messages are thrown. 

For example WPS 1.0.0 revision 0.3 with 4.1b throws the following:

```
Fatal Error: Error in call to extension function {public org.w3c.dom.NodeList 
com.occamlab.te.TECore.request(org.w3c.dom.Document,java.lang.String) throws 
java.lang.Throwable}: Exception in extension function java.lang.reflect.InvocationTargetException; 
SystemID: file:/Users/lbermudez/te-build/catalina_base/TE_BASE/work/_Users_lbermudez_te-
build_catalina_base_TE_BASE_scripts_wps_1.0.0_ctl_/wps/wps$general-WPS.General-
InvalidRequest.1.test; Line#: 77; Column#: -1


Fatal Error: Error in call to extension function {public org.w3c.dom.NodeList 
com.occamlab.te.TECore.request(org.w3c.dom.Document,java.lang.String) throws 
java.lang.Throwable}: Exception in extension function java.lang.reflect.InvocationTargetException; 
SystemID: file:/Users/lbermudez/te-build/catalina_base/TE_BASE/work/_Users_lbermudez_te-
build_catalina_base_TE_BASE_scripts_wps_1.0.0_ctl_/wps/wps$core-main.test; Line#: 118; 
Column#: -1


Fatal Error: Error in call to extension function {public synchronized void 
com.occamlab.te.TECore.callTest(net.sf.saxon.expr.XPathContext,java.lang.String,java.lang.String,net
.sf.saxon.om.NodeInfo,java.lang.String) throws java.lang.Exception}: Exception in extension function 
java.util.NoSuchElementException; SystemID: file:/Users/lbermudez/te-
build/catalina_base/TE_BASE/work/_Users_lbermudez_te-
build_catalina_base_TE_BASE_scripts_wps_1.0.0_ctl_/wps/wps$Main.test; Line#: 248; Column#: -1

```

**Comment by bermud** [comment]
Wait until WPS test is revised

**Comment by dstenger** [comment]
@keshav-nangare 
Can you please check if it is possible to improve error messages containing `Error in call to extension function`?

**Comment by keshav-nangare** [comment]
@dstenger 

I have tried to update this message but we cannot update only specific message because it's generated by the `SaxonApiException` class. If we updated the message then we lose the root cause information and file path information.

**Comment by keshav-nangare** [comment]
Fixed with PR #391.
